google/cloud/storage/oauth2/google_credentials_test.cc

```diff
@@ -215,11 +215,31 @@ TEST_F(GoogleCredentialsTest, LoadValidServiceAccountCredentialsFromFilename) {
   EXPECT_EQ(typeid(*ptr), typeid(ServiceAccountCredentials<>));
 }
 
+TEST_F(GoogleCredentialsTest,
+       LoadValidServiceAccountCredentialsFromFilenameWithOptionalArgs) {
+  std::string filename = ::testing::TempDir() + SERVICE_ACCOUNT_CRED_FILENAME;
+  SetupServiceAccountCredentialsFileForTest(filename);
+
+  // Test that the service account credentials are loaded from a file.
+  auto creds = CreateServiceAccountCredentialsFromJsonFilePath(
+      filename,
+      std::set<std::string>{
+          "https://www.googleapis.com/auth/devstorage.full_control"},
+      "user@foo.bar");
+  ASSERT_TRUE(creds.ok()) << "status=" << creds.status();
+  auto credentials = std::move(*creds);
+  auto ptr = credentials.get();
```

Comment:
Prefer `auto* ptr` over `auto ptr`. The explicit asterisk there is helpful for readers to know that the type is a pointer and will require indirection.

This style (`auto*`) also matches the required use for references. That is, a bare `auto` will always deduce to a value, not a reference. If you want to catch the value as a reference, you must qualify with the `&`. So, it's nice to follow a similar pattern with pointers and the asterisk.

Reply:
Neat! Done.